c10/core/TensorImpl.cpp

```diff
@@ -130,6 +130,35 @@ bool TensorImpl::compute_strides_like_channels_last() const {
   return false;
 }
 
+bool TensorImpl::compute_non_overlapping_and_dense() const {
+  if (dim() == 1) {
+    return size(0) < 2 || stride(0) == 1;
+  }
+  SmallVector<int64_t,5> perm;
+  perm.resize(dim());
+  for (int64_t i = 0; i < dim(); i ++) {
+    perm[i] = i;
+  }
+  // Sort by strides, leaving 0 and 1 sized dims at the end of the array
+  std::sort(perm.begin(), perm.end(), [&](int a, int b) {
+      if (sizes_[a] < 2) {
```

Comment:
do we actually want to allow strides 0? If someone created a tensor with stride 0 - isn't it overlapping?

Reply:
Strides 0 is not allowed, Size 0 is fine. But generally it is already considered is_contiguous by old code and density not going to be checked.